admin/spec/back-end/authentication/middleware.spec.js

```diff
@@ -0,0 +1,110 @@
+      spyOn(req, 'isAuthenticated').and.returnValue(true)
+      const func = isAuthenticated('SERVICE-MANAGER')
+      func(req, res, next)
+      expect(next).toHaveBeenCalled()
+    })
+    it('authenticates a user against an allowed list of roles', () => {
+      reqParams.user = {
+        id: 1,
+        UserName: 'UserName',
+        role: 'SERVICE-MANAGER'
+      }
+      const req = httpMocks.createRequest(reqParams)
+      spyOn(req, 'isAuthenticated').and.returnValue(true)
+      const allowedRoles = ['SERVICE-MANAGER', 'TEACHER']
+      const func = isAuthenticated(allowedRoles)
+      func(req, res, next)
+      expect(next).toHaveBeenCalled()
+    })
+    it('does not authenticate a user that is not matching a role of a string type', () => {
+      reqParams.user = {
+        id: 1,
+        UserName: 'UserName',
+        role: 'TEST-DEVELOPER'
+      }
+      const req = httpMocks.createRequest(reqParams)
+      spyOn(req, 'isAuthenticated').and.returnValue(true)
+      spyOn(res, 'redirect')
+      const allowedRole = 'TEACHER'
+      const func = isAuthenticated(allowedRole)
+      func(req, res, next)
+      expect(next).not.toHaveBeenCalled()
+      expect(res.redirect).toHaveBeenCalledWith('/unauthorised')
+    })
+    it('does not authenticate a user that is not included in the list of allowed roles', () => {
+      reqParams.user = {
+        id: 1,
+        UserName: 'UserName',
+        role: 'TEST-DEVELOPER'
+      }
+      const req = httpMocks.createRequest(reqParams)
+      spyOn(req, 'isAuthenticated').and.returnValue(true)
+      spyOn(res, 'redirect')
+      const allowedRoles = ['SERVICE-MANAGER', 'TEACHER']
+      const func = isAuthenticated(allowedRoles)
+      func(req, res, next)
+      expect(next).not.toHaveBeenCalled()
+      expect(res.redirect).toHaveBeenCalledWith('/unauthorised')
+    })
+    it('calls next if the role passed is undefined while the request is authenticated', () => {
+      reqParams.user = {
+        id: 1,
+        UserName: 'UserName',
+        role: 'TEST-DEVELOPER'
+      }
+      const req = httpMocks.createRequest(reqParams)
+      spyOn(req, 'isAuthenticated').and.returnValue(true)
+      spyOn(res, 'redirect')
+      const func = isAuthenticated(undefined)
+      func(req, res, next)
+      expect(next).toHaveBeenCalled()
+      expect(res.redirect).not.toHaveBeenCalled()
+    })
+  })
+  describe(' if the request is not authenticated', () => {
+    it('redirects to sign in route', () => {
+      reqParams.user = {
+        id: 1,
+        UserName: 'UserName',
+        role: 'TEST-DEVELOPER'
+      }
+      const req = httpMocks.createRequest(reqParams)
+      spyOn(req, 'isAuthenticated').and.returnValue(false)
+      spyOn(res, 'redirect')
+      const func = isAuthenticated(undefined)
+      func(req, res, next)
+      expect(next).not.toHaveBeenCalled()
+      expect(res.redirect).toHaveBeenCalledWith('/sign-in')
+    })
+  })
+})
```

Comment:
Should be a test where the userRole is not defined.

Reply:
@jon-shipley done